src/xtgeo/grid3d/grid_properties.py

```diff
@@ -79,6 +90,9 @@ def gridproperties_from_file(
     else:
         fformat = pfile.generic_format_by_proposal(fformat)  # default
 
+    if fformat is None:
+        raise RuntimeError("fformat is None")
```

Comment:
Is this to satisfy mypy? Only asking because `detect_fformat()` will return `"unknown"` (i.e. always a string) if nothing is found and `generic_format_by_proposal` will raise a ValueError if nothing is found, or always a string